### ISSUE TITLE
Error in Dockerfile: 'python' package not available, and 'curl' not found

### ISSUE THREAD

**Issue by ashutosji** [googleforgames/agones#3996]
**What happened**:  While building the Docker image, the following errors occurred:

1. The python package was not found, resulting in an error stating that 'python has no installation candidate.'
2. Later in the build process, the curl command was not recognized, leading to a non-zero exit code (127).

**What you expected to happen**: The Docker build process should have completed successfully with all specified dependencies (including python and curl) installed, allowing the subsequent commands in the Dockerfile to run without errors.

**How to reproduce it (as minimally and precisely as possible)**:

1. Go to the following [Dockerfile](https://github.com/googleforgames/agones/blob/main/build/e2e-image/Dockerfile).
2. Run the `docker build -t tag-name .` command to attempt to build the image.
3. Observe the errors related to the python and curl commands.

**Anything else we need to know?**:

**Environment**:
- Agones version: 
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Install method (yaml/helm):
- Troubleshooting guide log(s):
- Others:

**Comment by ashutosji** [comment]
> I cannot reproduce the issue with the latest code. No error thrown for `docker build -t tag-name .`. Can you double check whether it's reproducible?

Two pull requests (PRs) I reviewed yesterday failed in the CI pipeline with the same error message [here](https://pantheon.corp.google.com/cloud-build/builds/d65dacc1-5515-4245-96ac-7934a61f12e9?project=agones-images&e=13803378&mods=logs_tg_prod).
I attempted to reproduce the issue locally with the latest code, but also encountered the error. As a result, I filed a bug report.
However, upon rerunning the CI pipeline, the tests passed successfully. Additionally, I was unable to replicate the error locally anymore.
